src/ssl/ErrorDetailManager.cc

```diff
@@ -42,40 +63,11 @@ class ErrorDetailFile : public TemplateFile
 }// namespace Ssl
 
 /******************/
-bool
-Ssl::ErrorDetailsList::getRecord(Security::ErrorCode value, ErrorDetailEntry &entry)
+const Ssl::ErrorDetailEntry *
+Ssl::ErrorDetailsList::findRecord(Security::ErrorCode value) const
 {
     const ErrorDetails::const_iterator it = theList.find(value);
-    if (it != theList.end()) {
-        entry.error_no =  it->second.error_no;
-        entry.name =  it->second.name;
-        entry.detail =  it->second.detail;
-        entry.descr =  it->second.descr;
-        return true;
-    }
-    return false;
-}
-
-const char *
-Ssl::ErrorDetailsList::getErrorDescr(Security::ErrorCode value)
-{
-    const ErrorDetails::const_iterator it = theList.find(value);
-    if (it != theList.end()) {
-        return it->second.descr.termedBuf();
-    }
-
-    return nullptr;
-}
-
-const char *
-Ssl::ErrorDetailsList::getErrorDetail(Security::ErrorCode value)
```

Comment:
I have removed getErrorDescr() and getErrorDetail() (and the corresponding ErrorDetailsManager methods) because they had to be modified to work with SBuf _and_ these "find record and extract the requested field" methods were only making things more complex. An updated findRecord() interface addresses old caller needs without requiring a wrapper function for every field and "forwarding" those wrappers through the ErrorDetailsManager class to the ultimate consumer.

The diff does not show some of these modifications well, unfortunately.

Reply:
ok, thanks!